google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -339,9 +340,16 @@ public GapicSpannerRpc(final SpannerOptions options) {
                       // This sets the response compressor (Server -> Client).
                       .withEncoding(compressorName))
               .setHeaderProvider(headerProviderWithUserAgent)
+              .setAllowNonDefaultServiceAccount(true)
               // Attempts direct access to spanner service over gRPC to improve throughput,
               // whether the attempt is allowed is totally controlled by service owner.
-              .setAttemptDirectPath(true);
+              // We'll only attempt DirectPath if we are using real credentials.
+              // NoCredentials is used for plain text connections, for example when connecting to
+              // the emulator.
+              .setAttemptDirectPath(
+                  options.isAttemptDirectPath()
+                      && !Objects.equals(
+                          options.getScopedCredentials(), NoCredentials.getInstance()));
```

Comment:
`!Objects.equals(
                          options.getScopedCredentials(), NoCredentials.getInstance())`

This sounds more like a identification strategy for emulator. Can't we use the env variable used to identify emulator instead?

Reply:
We need this for more than just the emulator. This is also used for the in-memory Spanner mock server tests that also use `NoCredentials`. Also; It is possible to connect to the emulator without setting the environment variable. You can also just directly set the host name.